src/udev/udev-rules.c

```diff
@@ -1858,6 +1871,23 @@ static int udev_rule_apply_token_to_event(
         case TK_A_OPTIONS_DEVLINK_PRIORITY:
                 device_set_devlink_priority(dev, PTR_TO_INT(token->data));
                 break;
+        case TK_A_OPTIONS_LOG_LEVEL: {
+                int level = PTR_TO_INT(token->data);
+
+                if (level < 0)
+                        level = event->default_log_level;
+
+                log_set_max_level_realm(LOG_REALM_UDEV, level);
+                log_set_max_level_realm(LOG_REALM_SYSTEMD, level);
+
+                if (level == LOG_DEBUG && !event->log_level_was_debug) {
+                        /* The log level becomes LOG_DEBUG at first time. Let's log basic information. */
+                        log_device_uevent(dev, LOG_DEBUG, "Log level is changed to 'debug' while processing device");
+                        event->log_level_was_debug = true;
+                }
```

Comment:
I don't understand why the change in log level is reported only when switching to the DEBUG log level. Wouldn't it be less surprising if a change in log level would always be reported? It may also be worth to report the log level change for each event, not only the first time. I guess that the DEBUG log level is anyhow chatty enough that one more message is not going to matter much.

Reply:
> I don't understand why the change in log level is reported only when switching to the DEBUG log level. Wouldn't it be less surprising if a change in log level would always be reported? It may also be worth to report the log level change for each event, not only the first time. I guess that the DEBUG log level is anyhow chatty enough that one more message is not going to matter much.

The main motivation of this block is not reporting the log level change, but showing what uevent is currently processed. We usually show such information in DEBUG level. So, I do not think it is not necessary to log here in other level. When the main process is invoked with SYSTEMD_LOG_LEVEL=debug or something, then `worker_process_device()` logs the uevent information. However, if the log level of the main process is not DEBUG, then the information is not logged at the beginning of `worker_process_device()`, as at that time the maximum log level is not DEBUG. So, to obtain better debug logs, we should logs uevent info when the log level is changed to DEBUG in the first time.